### ISSUE TITLE
No header link anchor generated

### ISSUE THREAD

**Issue by Scal-Human** [SimonCropp/PandocNet#160]
Hi,

Searching to automate ebook production, I was planning to use your library over Pandoc to produce Html from Markdown (and then ePub).
The first test I make is a simple poem with a title which I was expecting to be marked (anchor) to have a link to the fragment.
The first line of the poem is the title header:
```
# Ce qu'on ne dit pas à propos de l'amour
``` 
Using the Pandoc command-line (v 3.2.1)
```powershell
pandoc "C:\...\Poem.md" -o "C:\...\Poem-Cmd.html"
``` 
the first line decomes:
```html
<h1 id="ce-quon-ne-dit-pas-à-propos-de-lamour">Ce qu’on ne dit pas à propos de l’amour</h1>
``` 
Using the Pandoc library (v 3.1.1)
```c#
PandocInstance.Convert<CommonMarkIn, HtmlOut>(@"C:\...\Poem.md", @"C:\...\Poem-Lib.html");
``` 
the first line of the result is
```html
<h1>Ce qu'on ne dit pas à propos de l'amour</h1>
``` 

FYI: I am using dotNet Sdk 8.0.303 targetting net8.0

I had a look in the parameters, but did not found anything related, and I am comparing the runs with the same "all defaults" for command-line and library.

**Comment by SimonCropp** [comment]
have you tried debuggin it and seeing if the command line parameters differ in any way?

**Comment by Scal-Human** [comment]
@SimonCropp No, and the command line executes too fast to get information from process explorer (even with 9s highlight).
I tried to set the <i>LogFile</i> in the <i>Options</i> but it writes an empty json "[]".
But note that Pandoc command line creates header id's without any parameter other than iput and output file.

**Comment by Scal-Human** [comment]
I added a test of my own in the tests project, and tryied to break in it but the breakpoint is not taken into account by VSCode.
The output result is the same, no ID on the h1 or h2 headers.
And without breakpoint, I cannot go down to the command-line construction.
But it is easily reproduced with markdown as
```
# Header text

The quick brown fox jumps over the lazy dog.

## Intro

Introdution.
``` 
and call like in test project
```c#
await PandocInstance.Convert<CommonMarkIn, HtmlOut>("sample.md", "output.html");
```

**Comment by SimonCropp** [comment]
CommonMark in pandoc does not seem to add the id. the same happens if i use the comand line

```
C:\Code>pandoc.exe a.md --from=commonmark --to=html
<h1>Header text</h1>
```

if u omit commonmark  then the default is actuall `PandocMdIn`

```
await PandocInstance.Convert<PandocMdIn, HtmlOut>("sample.md", "output.html");
```
